enhancements/cluster-api/installing-openshift-natively-via-cluster-api.md

```diff
@@ -0,0 +1,598 @@
+
+As an OpenShift developer, I want to simplify the maintenance of the installer by removing the embedded terraform and platform specific logic, so that the installer is easier to maintain and extend.
+
+As a Hive developer, I want to delegate as much of the cluster lifecycle to Cluster API as possible, so that Hive can focus on the OpenShift specific features.
+
+### Goals
+
+#### Phase 1
+
+1. Leverage an existing Cluster API control plane to provision infrastructure for OpenShift clusters.
+1. Delegate the provisioning and management of the cluster infrastructure bootstrap provisioning to Cluster API.
+1. Delegate the orchestration of the OpenShift cluster bootstrap node process to Cluster API.
+1. Supplement Cluster API infrastructure provisioning gaps with small, and focused supplemental infrastructure providers.
+1. Convert the installer to generate Cluster API infrastructure templates instead of terraform resources.
+1. Rely on the installer to generate configuration up to and including the `manifests` phase.
+1. Leverage the desired payload installer to generate bootstrap ignition configs and kubeconfigs.
+
+#### Phase 2
+
+1. Provide an end to end OpenShift installation experience using Cluster API.
+  1. This will allow a BYO infrastructure experience for OpenShift via Cluster API.
+1. Integrate Hive directly with Cluster API to remove the need to execute the installer during Cluster Deployments.
+
+### Non-Goals
+
+1. Provisioning of the Cluster API control plane.[^2]
+1. Generation in-tree of ignition data for the bootstrap node.
+
+## Proposal
+
+A Cluster API control plane and bootstrap provider will be created to handle the orchestration and configuration of the OpenShift cluster during the bootstrap process.
+The control plane provider will be responsible for creating (and destroying) the bootstrap node, and provisioning the control plane nodes once the bootstrap node is ready.
+The bootstrap provider will be responsible for generating the correct ignition data for the bootstrap node, control plane nodes and worker nodes
+
+Where gaps exist in the Cluster API infrastructure providers, small, focused supplemental infrastructure providers will be created to fill those gaps.
+For example, on AWS, where Cluster API does not manage DNS, a supplemental provider will be created to manage DNS records for the cluster.
+
+Provide a configurable, opinionated and controller driven multi-cluster and single cluster provisioning experience:
+* Driven by install config and Cluster API resources
+* Cluster API resources generated by the installer for opinionated install
+* Self crafted Cluster API resources for custom install or install on external platforms
+* Customisation of bootstrap manifests via control plane provider configuration
+
+### Workflow Description
+
+#### Opinionated installer generated infrastructure definitions
+
+The installer binary will be updated to transform the existing install config into Cluster API resources.
```

Comment:
This very much depends on the distro and where you're using CAPI. A lot of folks interact directly with the resources, especially those who are leveraging CAPI on their own infrastructure, like certain customers I'm aware of. Then there are people who have integrated it into their product, some of those have wrapped it. Eg Tanzu exposes an abstraction on top of CAPI resources rather than exposing the resources directly. So they will have something similar to this logic.